test/types/schema.test.ts

```diff
@@ -599,20 +600,6 @@ function gh11997() {
   userSchema.index({ name: 1 }, { weights: { name: 1 } });
 }
 
-function gh12003() {
```

Comment:
This test is incorrect, Auto typed schema will not work if typeKey has been changed in this way, because it will be "type" and will not reflect the correct type of the actual value.
@Uzlopak  @vkarpov15  do you have any thoughts?

Reply:
I don't think we can support this way to supply schema options with auto typed schema.
It will work just if the end user use "const" assertion and not use predefined type, then the type will reflect the actual value.